OpenRA.Mods.Common/Graphics/ActorPreview.cs

```diff
@@ -67,20 +67,35 @@ public Func<WRot> GetOrientation()
 			if (facingInfo == null)
 				return () => WRot.None;
 
+			WAngle? facing = null;
+
 			// Dynamic facing takes priority
 			var dynamicInit = reference.GetOrDefault<DynamicFacingInit>();
 			if (dynamicInit != null)
 			{
-				// TODO: Account for terrain slope
 				var getFacing = dynamicInit.Value;
-				return () => WRot.FromYaw(getFacing());
+				facing = getFacing();
 			}
 
-			// Fall back to initial actor facing if an Init isn't available
-			var facingInit = reference.GetOrDefault<FacingInit>();
-			var facing = facingInit != null ? facingInit.Value : facingInfo.GetInitialFacing();
-			var orientation = WRot.FromYaw(facing);
-			return () => orientation;
+			if (facing == null)
```

Comment:
```suggestion
			else
```

can't be null otherwise